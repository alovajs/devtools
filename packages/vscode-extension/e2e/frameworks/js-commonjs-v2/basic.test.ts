import sinon from 'sinon'
import { Commands, executeCommand, expect, getExt, is, MockWormhole, openFile, setupTest } from '../../ctx'

setupTest('js-commonjs-v2', () => {
  it('opens entry file', async () => {
    const packageText = await openFile('package.json')
    expect(packageText).to.include(`"type": "commonjs"`)
  })

  it('is active', () => {
    const ext = getExt()
    is(ext?.isActive, true)
  })
  it('createConfig', async () => {
    // Create spies and stubs
    sinon.define(MockWormhole, 'createConfig', () => {})
    const mockCreateConfig = sinon.stub(MockWormhole, 'createConfig').returns(Promise.resolve(undefined))

    // execute command
    await executeCommand(Commands.create_config)

    // Verify that generateConfig is called correctly
    expect(mockCreateConfig.calledOnce).to.equals(true)
  })
  it('createConfig content', async () => {
    await executeCommand(Commands.create_config)
    // Verify that generateConfig is called correctly
    expect(await openFile('alova.config.js')).to.matchSnapshot()
  })
})
