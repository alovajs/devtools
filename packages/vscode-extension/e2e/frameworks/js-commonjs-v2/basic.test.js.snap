// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`js-commonjs-v2 should works 1`] = `2`;
